route-snapper/src/lib.rs

```diff
@@ -290,6 +295,15 @@ impl JsRouteSnapper {
             );
 
             if let (Some(last), Waypoint::Snapped(current)) = (self.route.waypoints.last(), hover) {
+                let node1 = match last.to_owned() {
+                    Waypoint::Snapped(node) => node,
+                    Waypoint::Free(pt) => self.mouseover_node(pt).unwrap(),
+                };
+
+                if !is_connected(node1, current, &self.router) {
+                    return String::from("[]");
```

Comment:
In this case maybe we don't want to draw the current extension to the route (this big if block), but we still want to draw any confirmed waypoints and parts of the route